girder/test/test_size.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright 2013 Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+import collections
+import pytest
+
+from girder.models.collection import Collection
+from girder.models.file import File
+from girder.models.folder import Folder
+from girder.models.item import Item
+from girder.models.user import User
+from pytest_girder.assertions import assertStatus, assertStatusOk
+
+Hierarchy = collections.namedtuple('Hierarchy', ['collections', 'folders', 'items', 'files'])
+
+
+@pytest.fixture
+def hierarchy(admin, fsAssetstore):
+    c1 = Collection().createCollection(name='Coll1', creator=admin)
+    c2 = Collection().createCollection(name='Coll2', creator=admin)
+    f1 = Folder().createFolder(parent=c1, creator=admin, parentType='collection', name='Top level')
+    f2 = Folder().createFolder(parent=f1, creator=admin, parentType='folder', name='Subfolder')
+    i1 = Item().createItem(name='Item1', creator=admin, folder=f1)
+    i2 = Item().createItem(name='Item10', creator=admin, folder=f2)
+
+    file1 = File().createFile(
+        name='File1', creator=admin, item=i1, size=1, assetstore=fsAssetstore, mimeType='text/csv')
+    file1['sha512'] = ''  # TODO IS THIS NECESSARY?
+    File().save(file1, validate=False)
+
+    file2 = File().createFile(
+        name='File2', creator=admin, item=i2, size=10, assetstore=fsAssetstore, mimeType='text/csv')
+    file2['sha512'] = ''
+    File().save(file2, validate=False)
+
+    yield Hierarchy(collections=(c1, c2), folders=(f1, f2), items=(i1, i2), files=(file1, file2))
+
+    Collection().remove(c1)
+    Collection().remove(c2)
+
+
+def assertNodeSize(resource, model, size):
+    model = model().load(resource['_id'], force=True)
```

Comment:
Add `__tracebackhide__ = True` to fix the traceback for assertion functions.

https://docs.pytest.org/en/latest/example/simple.html#writing-well-integrated-assertion-helpers